### PR TITLE
add openapi ir

### PR DESCRIPTION
**Pull request by dsinghvi**

### CODE REVIEW

File: fern/openapi-ir/definition/ir.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+types: 
+  IntermediateRepresentation: 
+    properties: 
+      displayName: optional<string>
+      environments: list<Environment>
```
Comment (zachkirsch) [inline fern/openapi-ir/definition/ir.yml:5]: this is the `servers` block right? i'd just use their names, makes it easy to externalize if we want to

Comment (dsinghvi) [reply fern/openapi-ir/definition/ir.yml:5]: yup it is

File: fern/openapi-ir/definition/ir.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+  HttpMethod: 
+    enum: 
+      - GET
+      - POST
+      - PUT
+      - DELETE
+      - PATCH
+
+  Schema:
```
Comment (zachkirsch) [inline fern/openapi-ir/definition/ir.yml:78]: can Schemas have tags too?

File: fern/openapi-ir/definition/ir.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+  Environment: 
+    extends: WithDescription
+    properties: 
+      name: optional<string>
+      url: string
+
+  TagId: 
+    type: string
+    docs: The name of the tag.
```
Comment (zachkirsch) [inline fern/openapi-ir/definition/ir.yml:24]: `TagName` or `TagId`?

File: fern/openapi-ir/definition/ir.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  
+  TagInfo:
```
Comment (zachkirsch) [inline fern/openapi-ir/definition/ir.yml:26]: this isn't used

File: fern/openapi-ir/definition/ir.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    docs: The name of the tag.
+  
+  TagInfo: 
+    properties: 
+      description: string
+      displayName: optional<string>
+
+  SchemaId: 
+    type: string
+    docs: The name of the schema.
```
Comment (zachkirsch) [inline fern/openapi-ir/definition/ir.yml:33]: name or id? are these the same thing in openapi? would just pick one for clarity

File: fern/openapi-ir/definition/ir.yml (hunk 3 above)
Comment (dsinghvi) [reply fern/openapi-ir/definition/ir.yml:24]: it's a unique name so wasn't sure -- decided to call them ids (see `SchemaId` but i can switch)

File: fern/openapi-ir/definition/ir.yml (hunk 2 above)
Comment (dsinghvi) [reply fern/openapi-ir/definition/ir.yml:78]: nope

Comment (dsinghvi) [reply fern/openapi-ir/definition/ir.yml:78]: not directly atleast

File: fern/openapi-ir/definition/ir.yml (hunk 4 above)
Comment (dsinghvi) [reply fern/openapi-ir/definition/ir.yml:26]: will add later

**Pull request closed without merging**